pkg/resources/envoy/deployment.go

```diff
@@ -58,32 +69,35 @@ func (r *Reconciler) deployment(log logr.Logger, extListener v1beta1.ExternalLis
 	}
 
 	return &appsv1.Deployment{
-		ObjectMeta: templates.ObjectMeta(
-			fmt.Sprintf(envoyDeploymentName, extListener.Name, r.KafkaCluster.GetName()),
-			labelsForEnvoyIngress(r.KafkaCluster.GetName(), extListener.Name), r.KafkaCluster),
+		ObjectMeta: templates.ObjectMetaWithAnnotations(
+			deploymentName,
+			labelsForEnvoyIngress(r.KafkaCluster.GetName(), annotationName),
```

Comment:
Why `annotationName` is passed into `labelsForEnvoyIngress` instead of `extListener.Name`?